### PR TITLE
Separates AppendText for MessagesLog. Adds process memory limit on Me…

### PR DESCRIPTION
**Pull request by HelioGuilherme66**
…ssagesLog.

Possible fix for #1687

### CODE REVIEW

File: src/robotide/contrib/testrunner/testrunnerplugin.py [hunk 1]
```diff
@@ -70,7 +72,7 @@
 from robotide.robotapi import LOG_LEVELS
 from robotide.utils import robottime, is_unicode, PY2
 from robotide.preferences.editors import ReadFonts
-from sys import getfilesystemencoding
+from sys import getfilesystemencoding, getsizeof
```
Comment (Nyral) [inline src/robotide/contrib/testrunner/testrunnerplugin.py]: `getsizeof` is not used

**Comment by Nyral** [comment]
It still crashes for me.

**Comment by HelioGuilherme66** [comment]
Using the memory_profile tool, we see that on the AppendText with encoding it takes about 200-500KB of memory increase. That is why I have isolated the messages_log. But I did not changed the output window. Also if you run from command line, the most of memory use is in the end when processing output.xml to get log and report. Don't really know what else we can do, other then suggesting running `robot` outside of RIDE.

**Comment by Nyral** [comment]
Try running the test 2-3 times ;)

Your solution prevents text from added to memory log, but python ram usage still increases.
Console log is affected too, just use `Log to console` keyword instead.

I've noticed that ram usage is freed if you uncheck `Show message log`. That's because it destroys the object. Maybe we can try that? If memory usage increases too much, then destroy and recreate the textctrl?

**Comment by HelioGuilherme66** [comment]
Experimenting memory_profile (run all 1000 iteractions, but computer got slow when log/report generation):
`PYTHONPATH=/usr/local/lib/python3.7/site-packages mprof run src/robotide/__init__.py `

This is the chart generated by `mprof plot`:
image

**Comment by Nyral** [comment]
With the changes in latest commit, the memory limit no longer triggers.

**Comment by HelioGuilherme66** [comment]
It triggers, writes message on log, but is too fast to be noticed, because new messages are logged.

**Comment by Nyral** [comment]
Did you try running the MemoryLimit test again after you made those changes? RIDE crashes.

**Comment by HelioGuilherme66** [comment]
This PR is going to be on hold, because there is no noticeable advantage in merging it. Below are the charts for the test runs (3 consecutive runs with counter at 200). We can see a 500MB increase on each run. Further code changes must reduce that 500MB to a value near 0.
Before PR (master branch):
memory_before_PR

After PR (memdebug branch):
memory_with_PR

Was tested on Windows 10, python 3.7.4 wxPython 4.0.7.post2.

**Pull request merged**